sdk/core/Azure.Core.Experimental/src/ContentType.cs

```diff
@@ -0,0 +1,87 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+
+namespace Azure.Core
+{
+    /// <summary>
+    /// Represents content type.
+    /// </summary>
+    public readonly struct ContentType : IEquatable<ContentType>, IEquatable<string>
+    {
+        private readonly string _contentType;
+
+        /// <summary>
+        /// application/json
+        /// </summary>
+        public static ContentType ApplicationJson { get; } = new ContentType("application/json");
+
+        /// <summary>
+        /// application/json
+        /// </summary>
+        public static ContentType ApplicationOctetStream { get; } = new ContentType("application/octet-stream");
+
+        /// <summary>
+        /// application/json
+        /// </summary>
+        public static ContentType TextPlain { get; } = new ContentType("text/plain");
+
```

Comment:
Scrolling through https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types/Common_types, it might also be worth adding XML because it's still used by a few services like Storage.

Reply:
Will we ever have untyped (RequestContent, Stream, etc.) APIs that send XML?